src/libraries/System.Linq/src/System/Linq/Concat.SpeedOpt.cs

```diff
@@ -326,19 +326,21 @@ private TSource[] PreallocatingToArray()
 
             public override TSource? TryGetLast(out bool found)
             {
-                ConcatNIterator<TSource>? node = this;
+                ConcatNIterator<TSource>? node, previousN = this;
```

Comment:
Presumably you're duplicating locals to avoid introducing more branching inside the loop?

Reply:
My main motivator was to keep in consistency with the `GetEnumerable` method 
https://github.com/dotnet/runtime/blob/f24eb8a73e5776cf086dccb72737621f12232d39/src/libraries/System.Linq/src/System/Linq/Concat.cs#L179

It was probably chosen like this as you'd end up either having to accessing the `_tail` field multiple times or have some ugly code, but again I just copied the existing approach.